web3/eth.py

```diff
@@ -294,15 +294,22 @@ def call(self, transaction, block_identifier=None):
             [transaction, block_identifier],
         )
 
-    def estimateGas(self, transaction):
+    def estimateGas(self, transaction, block_identifier=None):
         # TODO: move to middleware
         if 'from' not in transaction and is_checksum_address(self.defaultAccount):
             transaction = assoc(transaction, 'from', self.defaultAccount)
 
-        return self.web3.manager.request_blocking(
-            "eth_estimateGas",
-            [transaction],
-        )
+        # TODO: move to middleware
+        if block_identifier is None:
```

Comment:
This code can be reduced to:
```python
params = [transaction]
if block_identifier:
    params.append(block_identifier)
return self.web3.manager.request_blocking(
    "eth_estimateGas",
    params
)
```

Reply:
Rather than mutating the params list, I would go with:

```py
        if block_identifier is None:
            params = [transaction]
        else:
            params = [transaction, block_identifier]

        return self.web3.manager.request_blocking(
            "eth_estimateGas",
            params,
        )
```

Also, we don't need the second `move to middleware` TODO.